libraries/provider_rightscale_volume.rb

```diff
@@ -556,7 +556,15 @@ def attach_volume(volume_id, device)
         scan_for_attachments if node['virtualization']['system'] == 'vmware'
 
         # Determine the actual device name
-        (Set.new(get_current_devices) - current_devices).first
+        added_device = nil
+        Timeout::timeout(60) do
+          begin
+            sleep(1)
+            Chef::Log.info "Checking for added device."
+            added_device = (Set.new(get_current_devices) - current_devices).first
+          end until !added_device.nil?
+        end
```

Comment:
Would a while loop be equivalent functional-wise?  I would favor it for two reasons.
1.  Puts the condition up at the top of the loop, more readable to me.
2.  Does away with the ! operator.
But check me on this, maybe they are not equivalent.

```ruby
Timeout::timeout(60) do
  while added_device.nil?
     sleep(1)
     Chef::Log.info "Checking for added device."
     added_device = (Set.new(get_current_devices) - current_devices).first
  end
end
```